android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateManager.java

```diff
@@ -223,67 +223,70 @@ public void downloadPackage(JSONObject updatePackage, String expectedBundleFileN
             String diffManifestFilePath = CodePushUtils.appendPathComponent(unzippedFolderPath,
                     CodePushConstants.DIFF_MANIFEST_FILE_NAME);
             boolean isDiffUpdate = FileUtils.fileAtPathExists(diffManifestFilePath);
+
             if (isDiffUpdate) {
                 String currentPackageFolderPath = getCurrentPackageFolderPath();
                 CodePushUpdateUtils.copyNecessaryFilesFromCurrentPackage(diffManifestFilePath, currentPackageFolderPath, newUpdateFolderPath);
                 File diffManifestFile = new File(diffManifestFilePath);
                 diffManifestFile.delete();
             }
 
-            FileUtils.copyDirectoryContents(unzippedFolderPath, newUpdateFolderPath);
-            FileUtils.deleteFileAtPathSilently(unzippedFolderPath);
-
             // For zip updates, we need to find the relative path to the jsBundle and save it in the
             // metadata so that we can find and run it easily the next time.
-            String relativeBundlePath = CodePushUpdateUtils.findJSBundleInUpdateContents(newUpdateFolderPath, expectedBundleFileName);
-
+            String relativeBundlePath = CodePushUpdateUtils.findJSBundleInUpdateContents(unzippedFolderPath, expectedBundleFileName);
             if (relativeBundlePath == null) {
-                throw new CodePushInvalidUpdateException("Update is invalid - A JS bundle file named \"" + expectedBundleFileName + "\" could not be found within the downloaded contents. Please check that you are releasing your CodePush updates using the exact same JS bundle file name that was shipped with your app's binary.");
-            } else {
-                if (FileUtils.fileAtPathExists(newUpdateMetadataPath)) {
-                    File metadataFileFromOldUpdate = new File(newUpdateMetadataPath);
-                    metadataFileFromOldUpdate.delete();
+                FileUtils.deleteFileAtPathSilently(unzippedFolderPath);
+                if (FileUtils.fileAtPathExists(newUpdateFolderPath)) {
+                    FileUtils.deleteFileAtPathSilently(newUpdateFolderPath);
```

Comment:
It seems that this deletion is unnecessary here as the same is done on lines 148-151 and the folder isn't created later till `FileUtils.copyDirectoryContents(unzippedFolderPath, newUpdateFolderPath)` (your 245 line)

Reply:
In case of `isDiffUpdate` true , it may have been created. So the conditional to check if `newUpdateFolderPath` exists followed by delete may be necessary to make sure. Please let me know.

https://github.com/Microsoft/react-native-code-push/pull/1367/files/1378ed54ab5c73dafcd2a49d17974db26515b357#diff-4aa56c56a1afb536396acce24d2d2015R227